### PR TITLE
CI Improvements

### PR DESCRIPTION
**Pull request by GrantBirki**
# CI Improvements

Greetings 👋!

I use the `cloudflare/wrangler-action` in all sorts of open source projects and wanted to help contribute now that I see some more activity in this repo.

I am a maintainer myself of several open source Actions. I created the [`github/branch-deploy`](https://github.com/github/branch-deploy) Action and wired up a workflow to fix a common problem that I saw happen a few times today in this repo. A maintainer will make changes, push changes, and then see CI unexpected fail... at first you think its because your tests are busted, but then you realize you just forgot to run `npm run build`. The `check-dist` workflow bundled in this PR will solve that for you

**TL;DR**:

- New `check-dist` workflow that ensures you ran `npm run build` before merging a pull request and landing "non-built" changes on `main`
- New dedicated `lint` workflow to help break apart CI jobs a bit
- Adding workflow status badges to the README
- I also ran `npm install` upon cloning and it changes the version in the lock file FYI. Should be fine though

### CODE REVIEW

**Comment by JacobMGEvans** [comment]
Hi, I plan on implementing an automated build & tag workflow w/ changesets! I am not sure how this PR will play into that yet but I will keep it up in case it aligns well with what I have planned 😄 

I am super happy to hear this has reinvigorated the OSS contributions in the repo that was my biggest hope ❤️

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]
Blocked for now.

**Comment by JacobMGEvans** [comment]
With these 2 PRs in for automating
This one for build & tagging for release:
https://github.com/cloudflare/wrangler-action/pull/128

This one for automating the release & changelog
https://github.com/cloudflare/wrangler-action/pull/132

I think this PR won't be needed, I appreciate the effort and if you have anymore ideas or improvements they are always welcomed! 😄
